src/plugins/intel_gpu/src/kernel_selector/cl_kernels/select_gpu_ref.cl

```diff
@@ -45,7 +45,11 @@ KERNEL(select)(
     uint output_offset = OUTPUT_GET_INDEX(b, f, y, x);
 #endif
 
-    const OUTPUT_TYPE res = TO_OUTPUT_TYPE(select(INPUT_2, INPUT_1, MASK));
+    #if INPUT1_IS_FP && !OUTPUT_IS_FP
+     const OUTPUT_TYPE res = TO_OUTPUT_TYPE(convert_long(select(INPUT_2, INPUT_1, MASK)));
```

Comment:
Why do you convert to `long` ? 64-bit types usually have poor perf, so you should avoid using it w/o real need.

Reply:
I refer to this code.
https://github.com/openvinotoolkit/openvino/blob/master/src/plugins/intel_gpu/src/kernel_selector/cl_kernels/reorder_data.cl#L216